src/metabase/task/truncate_audit_tables.clj

```diff
@@ -0,0 +1,91 @@
+  (:require
+   [clojurewerkz.quartzite.jobs :as jobs]
+   [clojurewerkz.quartzite.schedule.cron :as cron]
+   [clojurewerkz.quartzite.triggers :as triggers]
+   [java-time.api :as t]
+   [metabase.config :as config]
+   [metabase.models.setting :as setting]
+   [metabase.models.setting.multi-setting
+    :refer [define-multi-setting-impl]]
+   [metabase.models.task-history :as task-history]
+   [metabase.plugins.classloader :as classloader]
+   [metabase.public-settings.premium-features :as premium-features :refer [defenterprise]]
+   [metabase.task :as task]
+   [metabase.task.truncate-audit-tables.interface
+    :as truncate-audit-tables.i]
+   [metabase.util.log :as log]
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+;; Load EE implementation if available
+(when config/ee-available?
+  (classloader/require 'metabase-enterprise.task.truncate-audit-tables))
+
+(define-multi-setting-impl truncate-audit-tables.i/audit-max-retention-days :oss
+  :getter (fn []
+            (if-not (premium-features/is-hosted?)
+              ##Inf
+              (let [env-var-value      (setting/get-value-of-type :integer :audit-max-retention-days)
+                    min-retention-days truncate-audit-tables.i/min-retention-days]
+                (cond
+                  ((some-fn nil? zero?) env-var-value) ##Inf
+                  (< env-var-value min-retention-days) (do
+                                                         (truncate-audit-tables.i/log-minimum-value-warning env-var-value)
+                                                         min-retention-days)
+                  :else                                env-var-value)))))
+
+(defn- truncate-table!
+  "Given a model, deletes all rows older than the configured threshold"
+  [model time-column]
+  (when-not (infinite? (truncate-audit-tables.i/audit-max-retention-days))
+    (let [table-name (name (t2/table-name model))]
+      (task-history/with-task-history {:task "task-history-cleanup"}
+        (try
+          (log/infof "Cleaning up %s table" table-name)
+          (let [rows-deleted (t2/delete!
+                              model
+                              time-column
+                              [:<= (t/minus (t/offset-date-time) (t/days (truncate-audit-tables.i/audit-max-retention-days)))])]
+            (if (> rows-deleted 0)
+              (log/infof "%s cleanup successful, %d rows were deleted" table-name rows-deleted)
+              (log/infof "%s cleanup successful, no rows were deleted" table-name)))
+          (catch Throwable e
+            (log/errorf e "%s cleanup failed" table-name)))))))
+
+(defenterprise audit-models-to-truncate
+  "List of models to truncate. OSS implementation only truncates `query_execution` table."
+  metabase-enterprise.task.truncate-audit-tables
+  []
+  {:model/QueryExecution :started_at})
+
+(defn- truncate-audit-tables!
+  []
+  (dorun
```

Comment:
I think it's better to use `run!` rather than the `dorun (map ...` style here.

```clojure
(map
  (fn [[model time-column]]
    (truncate-table! model time-column))
  (audit-models-to-truncate))

;; vs

(run! (fn [[model time-column]]
        (truncate-table! model time-column))
      (audit-models-to-truncate))
```

Reply:
fair, I always forget about `run!`, thanks